spec/requests/evalutations_controller_spec.rb

```diff
@@ -152,4 +152,26 @@
       expect(response).to have_http_status :ok
     end
   end
+
+  describe '#copy_previous_principles' do
+    include_context "application season"
+    include_context "current user is the group leader"
+
+    before(:each) do
+      evaluation.update!(id:2)
+      previous_evaluation = create(:evaluation, group: group, id: 1 )
+      Principle.create!(evaluation: previous_evaluation,
+                        name: 'Previous principle',
+                        type: Principle::WORK,
+                        max_per_member: 10)
+    end
+
+    it 'copies the previous principles' do
+      expect {
+        post "/groups/#{group.id}/evaluations/#{evaluation.id}/copy_previous_principles"
+      }.to change{evaluation.principles.count}.from(0).to(1)
+
+      expect(response).to redirect_to(group_evaluation_principles_path(group, evaluation))
+    end
```

Comment:
What do you think about this? You can avoid a DB update by redefining the `let(:evaluation)`, and ensure the behaviour by verifying the new principle.

```suggestion
    let(:evaluation) { create(:evaluation, id: 2) }

    it 'copies the principles of the previous evaluation' do
      previous_evaluation = create(:evaluation, group: evaluation.group, id: 1)
      previous_principle  = create(:principle, evaluation: previous_evaluation)

      expect { post "/groups/#{group.id}/evaluations/#{evaluation.id}/copy_previous_principles" }
        .to change(evaluation.principles, :count).from(0).to(1)

      expect(response).to redirect_to(group_evaluation_principles_path(group, evaluation))
      expect(Principle.last)
        .to have_attributes(previous_principle.slice(:name, :description, :type, :max_per_member))
    end
```